packages/api/lib/granule-remove-from-cmr.js

```diff
@@ -1,39 +1,37 @@
-const { GranuleNotPublished, RecordDoesNotExist } = require('@cumulus/errors');
+const { GranuleNotPublished } = require('@cumulus/errors');
 const { CMR } = require('@cumulus/cmr-client');
 const log = require('@cumulus/common/log');
-const { CollectionPgModel, GranulePgModel } = require('@cumulus/db');
+const { getGranuleCollectionId, GranulePgModel } = require('@cumulus/db');
 const cmrjsCmrUtils = require('@cumulus/cmrjs/cmr-utils');
-const { deconstructCollectionId } = require('./utils');
-
 const models = require('../models');
 
 /**
  * Remove granule record from CMR
  *
- * @param {Object} granule - A granule record
+ * @param {Object} granule - A postgres granule record
+ * @param {string} collectionId - The CMR collection 'id' for the granule to be removed
  * @throws {GranuleNotPublished|Error}
  * @private
  */
-const _removeGranuleFromCmr = async (granule) => {
-  log.info(`granules.removeGranuleFromCmrByGranule ${granule.granuleId}`);
-
-  if (!granule.published || !granule.cmrLink) {
-    throw new GranuleNotPublished(`Granule ${granule.granuleId} is not published to CMR, so cannot be removed from CMR`);
+const _removeGranuleFromCmr = async (granule, collectionId) => {
+  log.info(`granules.removeGranuleFromCmrByGranule ${granule.granule_id}`);
```

Comment:
This should include the collectionId if we're using it in the selection

Reply:
👍 

https://github.com/nasa/cumulus/pull/2415/commits/b776e9065d06090709a69742eefab1b3cb14aa98